### PR TITLE
Remove legacy resend password reset

### PR DESCRIPTION
**Pull request by jfsoul**
## What does this change?
Removes usage of non-functional resend password reset page and replaces with a new version in identity frontend https://github.com/guardian/identity-frontend/pull/365

Further work required in a subsequent PR to remove some more potentially unused routes/code.  There looks to be a lot of password reset related code which I am doubtful is all required any more.

## What is the value of this and can you measure success?
Resend password reset works again.

### CODE REVIEW

**Review by mariogalic — APPROVED** [review]

File: identity/test/controllers/ResetPasswordControllerTest.scala [hunk 1]
```diff
@@ -39,6 +39,9 @@ class ResetPasswordControllerTest
   )
 
   when(requestParser.apply(MockitoMatchers.anyObject())).thenReturn(identityRequest)
+  when(idUrlBuilder.buildUrl(MockitoMatchers.eq("/reset/resend"), MockitoMatchers.eq(identityRequest), MockitoMatchers.anyVararg[(String, String)]))
+    .thenReturn("https://profile.thegulocal.com/reset/resend")
```
Comment (walaura) [inline identity/test/controllers/ResetPasswordControllerTest.scala]: `thegulocal `

File: identity/test/controllers/ResetPasswordControllerTest.scala [hunk 2]
```diff
@@ -68,7 +71,8 @@ class ResetPasswordControllerTest
       "should render to the the to the request new password form" in Fake {
         val result = resetPasswordController.processUpdatePasswordToken("1234")(fakeRequest)
         status(result) should equal(SEE_OTHER)
-        header("Location", result).head should be ("/requestnewtoken")
+        println(headers(result))
```
Comment (Calum-Campbell) [inline identity/test/controllers/ResetPasswordControllerTest.scala]: println

File: identity/test/controllers/ResetPasswordControllerTest.scala [hunk 3]
```diff
@@ -90,11 +94,13 @@ class ResetPasswordControllerTest
     }
 
     "when the reset token has expired" - {
+
       when(api.resetPassword("1234","newpassword")).thenReturn(Future.successful(Left(tokenExpired)))
       "should redirect to request request new password with a token expired" in Fake {
         val result = resetPasswordController.resetPassword("1234")(fakeRequest)
         status(result) should equal(SEE_OTHER)
-        header("Location", result).head should be ("/requestnewtoken")
+        println(headers(result))
```
Comment (Calum-Campbell) [inline identity/test/controllers/ResetPasswordControllerTest.scala]: println

**Review by walaura — COMMENTED** [review]
thrilled with anything that moves us closer to migrating away from frontend

**Pull request merged**